src/newmagic.cpp

```diff
@@ -1739,6 +1739,32 @@ void cast_health_spell(struct char_data *ch, int spell, int sub, int force, char
       } else
         send_to_char(FAILED_CAST, ch);
       break;
+    case SPELL_FLAME_AURA:
+      if (!check_spell_victim(ch, vict, spell, arg))
+        return;
+
+      if (would_become_killer(ch, vict)) {
+        send_to_char("That would make you a PLAYER KILLER! Both you and your opponent must `toggle PK` to do that.\r\n", ch);
+        return;
+      }
+
+      if (!IS_NPC(ch) && !IS_NPC(vict) && PLR_FLAGGED(ch, PLR_KILLER)) {
+        act("You have the KILLER flag, so you can't affect $N with an aggressive spell.", TRUE, ch, 0, vict, TO_CHAR);
+        return;
+      }
+      WAIT_STATE(ch, (int) (SPELL_WAIT_STATE_TIME));
+
+      success = success_test(skill, 4 + target_modifiers);
+      if (success > 0 && !AFF_FLAGGED(ch, AFF_FLAME_AURA)) {
```

Comment:
Checking to see if they already have the aff should come before this, and should give a specific message, like "They're already affected by that spell." See improved invis for an example of how this is done.

Reply:
Gotcha, check/message added.